src/libais/ais8_367.cpp

```diff
@@ -274,5 +274,459 @@ ostream& operator<< (ostream &o, const Ais8_367_25 &msg) {
   return o << msg.mmsi << ": " << msg.version << ": " << msg.utc_hour << ": " << msg.utc_min << ": " << msg.pressure << ": " << msg.wind_speed << ": " << msg.wind_dir;
 }
 
+// Environmental Message
+Ais8_367_33_Location::Ais8_367_33_Location(const AisBitset &bits,
+                                           const size_t offset) {
+  version = bits.ToUnsignedInt(offset, 6);
+  position = bits.ToAisPoint(offset + 6, 55);
+
+  precision = bits.ToUnsignedInt(offset + 61, 3);
+
+  altitude_raw = bits.ToUnsignedInt(offset + 64, 12);
+  altitude = altitude_raw / 10.;  // meters
+
+  owner = bits.ToUnsignedInt(offset + 76, 4);
+  timeout = bits.ToUnsignedInt(offset + 80, 3);
+  spare2 = bits.ToUnsignedInt(offset + 83, 2);
+}
+
+Ais8_367_33_Station::Ais8_367_33_Station(const AisBitset &bits,
+                                         const size_t offset) {
+  name = bits.ToString(offset, 84);
+  // Remove any trailing '@' characters.  Spec says they are not to be shown in the presentation.
+  while (name.back() == '@') {
+    name.pop_back();
```

Comment:
https://www.cplusplus.com/reference/string/string/back/ says:

`This function shall not be called on empty strings.`

While it currently works with clang and gcc there needs to be an empty check first.

```
while (!name.empty() && name.back() == '@') {
```

https://godbolt.org/z/1cPYoM8GT